internal/fsutil/create_test.go

```diff
@@ -62,37 +67,101 @@ var createTests = []createTest{{
 		"/tmp/": "dir 01775",
 	},
 }, {
+	summary: "Cannot create a parent directory without MakeParents set",
 	options: fsutil.CreateOptions{
 		Path: "foo/bar",
 		Mode: fs.ModeDir | 0775,
 	},
-	error: `.*: no such file or directory`,
+	error: `mkdir \/[^ ]*\/foo/bar: no such file or directory`,
 }, {
+	summary: "Re-creating an existing directory keeps the original mode",
 	options: fsutil.CreateOptions{
 		Path: "foo",
 		Mode: fs.ModeDir | 0775,
 	},
-	hackdir: func(c *C, dir string) {
-		c.Assert(os.Mkdir(filepath.Join(dir, "foo/"), fs.ModeDir|0765), IsNil)
+	hackopt: func(c *C, targetDir string, options *fsutil.CreateOptions) {
+		c.Assert(os.Mkdir(filepath.Join(targetDir, "foo/"), fs.ModeDir|0765), IsNil)
 	},
 	result: map[string]string{
 		// mode is not updated.
 		"/foo/": "dir 0765",
 	},
 }, {
+	summary: "Re-creating an existing file keeps the original mode",
 	options: fsutil.CreateOptions{
 		Path: "foo",
 		// Mode should be ignored for existing entry.
 		Mode: 0644,
 		Data: bytes.NewBufferString("changed"),
 	},
-	hackdir: func(c *C, dir string) {
-		c.Assert(os.WriteFile(filepath.Join(dir, "foo"), []byte("data"), 0666), IsNil)
+	hackopt: func(c *C, targetDir string, options *fsutil.CreateOptions) {
+		c.Assert(os.WriteFile(filepath.Join(targetDir, "foo"), []byte("data"), 0666), IsNil)
 	},
 	result: map[string]string{
 		// mode is not updated.
 		"/foo": "file 0666 d67e2e94",
 	},
+}, {
+	summary: "Create a hard link",
+	options: fsutil.CreateOptions{
+		Path:        "dir/link-to-file",
```

Comment:
```suggestion
		Path:        "dir/hardlink",
```

Reply:
Fixed in [03ce4d3](https://github.com/letFunny/chisel/pull/17/commits/03ce4d322ccaff4f6ba8c00a32aee039de12c9d3).